modules/gke-nodepool/main.tf

```diff
@@ -130,6 +155,21 @@ resource "google_container_node_pool" "nodepool" {
     content {
       max_surge       = try(var.nodepool_config.upgrade_settings.max_surge, null)
       max_unavailable = try(var.nodepool_config.upgrade_settings.max_unavailable, null)
+      strategy        = try(var.nodepool_config.upgrade_settings.strategy, "SURGE")
+      dynamic "blue_green_settings" {
+        for_each = try(var.nodepool_config.upgrade_settings.blue_green_settings, null) != null ? [""] : []
+        content {
+          node_pool_soak_duration = try(var.nodepool_config.upgrade_settings.blue_green_settings.node_pool_soak_duration, null)
```

Comment:
The try here is no necessary (you `var.nodepool_config.upgrade_settings.blue_green_settings` know  is not null)